rocketpy/Rocket.py

```diff
@@ -689,74 +696,50 @@ def finNumCorrection(n):
                 + (1 / 6) * (Cr + Ct - Cr * Ct / (Cr + Ct))
             )
 
-        # Calculate lift parameters for planar fins
         if not airfoil:
-
-            clalphaSingleFin = Function(
-                lambda mach: 2
-                * np.pi
-                * AR
-                * (Af / Aref)
-                / (2 + np.sqrt(4 + ((beta(mach) * AR) / (np.cos(gamac))) ** 2)),
+            # Defines clalpha2D as 2*pi for planar fins
+            clalpha2D = Function(lambda mach: 2 * np.pi / beta(mach))
+        else:
+            # Defines clalpha2D as the derivative of the
+            # lift coefficient curve for a specific airfoil
+            airfoilCl = Function(
+                airfoil[0],
+                interpolation="linear",
             )
 
-            clalphaMultipleFins = (
-                liftInterferenceFactor * finNumCorrection(n) * clalphaSingleFin
-            )  # Function of mach number
-
-            # Calculates clalpha * alpha
-            cl = Function(
-                lambda alpha, mach: alpha * clalphaMultipleFins(mach),
-                ["Alpha (rad)", "Mach"],
-                "Cl",
+            # Convert to radians if needed
+            if airfoil[1] == "degrees":
+                if callable(airfoil[0]):
+                    airfoilCl *= np.pi / 180
+                else:
+                    for i in range(len(airfoilCl)):
+                        airfoilCl[i][0] *= np.pi / 180
+
+            # Differentiating at x = 0 and correctign for compressible flow
+            clalpha2D = Function(
+                lambda mach: airfoilCl.differentiate(x=1e-3, dx=1e-3) / beta(mach)
```

Comment:
The current implementation can be inefficient, since every time `clalpha2D` is called, the function `airfoilCl` is differentiated. Since the result will always be the same, it makes sense to do the differentiation only once.

Furthermore, we can also improbe the current degrees to radians correction to all values of the `airfoilCl` Function. What if we differentiate first, and then convert only the derivative result, instead of multiple values?

Here is my suggestion:
```suggestion
            # Differentiating at x = 0 to get cl_alpha
            clalpha2D_Mach0 = airfoilCl.differentiate(x=1e-3, dx=1e-3)

            # Convert to radians if needed
            if airfoil[1] == "degrees": clalpha2D_Mach0 *= 180/np.pi

            # Correcting for compressible flow
            clalpha2D = Function(
                lambda mach: clalpha2D_Mach0 / beta(mach)
```

Reply:
I really like this! Seems to work perfectly too